src/ui/ViewModelBase.hh

```diff
@@ -0,0 +1,101 @@
+    void RemoveNotifyTarget(NotifyTarget& pTarget) { m_vNotifyTargets.erase(&pTarget); }
+
+protected:
+    ViewModelBase() = default;
+
+    /// <summary>
+    /// Gets the value associated to the requested boolean property.
+    /// </summary>
+    /// <param name="pProperty">The property to query.</param>
+    /// <returns>The current value of the property for this object.</returns>
+    bool GetValue(const BoolModelProperty& pProperty) const
+    {
+        IntModelProperty::ValueMap::const_iterator iter = m_mIntValues.find(pProperty.GetKey());
+        return static_cast<bool>(iter != m_mIntValues.end() ? iter->second : pProperty.GetDefaultValue());
+    }
+
+    /// <summary>
+    /// Sets the specified boolean property to the specified value.
+    /// </summary>
+    /// <param name="pProperty">The property to set.</param>
+    /// <param name="bValue">The value to set.</param>
+    void SetValue(const BoolModelProperty& pProperty, bool bValue);
+
+    /// <summary>
+    /// Gets the value associated to the requested string property.
+    /// </summary>
+    /// <param name="pProperty">The property to query.</param>
+    /// <returns>The current value of the property for this object.</returns>
+    const std::wstring& GetValue(const StringModelProperty& pProperty) const
+    {
+        StringModelProperty::ValueMap::const_iterator iter = m_mStringValues.find(pProperty.GetKey());
+        return (iter != m_mStringValues.end() ? iter->second : pProperty.GetDefaultValue());
+    }
+
+    /// <summary>
+    /// Sets the specified string property to the specified value.
+    /// </summary>
+    /// <param name="pProperty">The property to set.</param>
+    /// <param name="sValue">The value to set.</param>
+    void SetValue(const StringModelProperty& pProperty, const std::wstring& sValue);
+
+    /// <summary>
+    /// Gets the value associated to the requested integer property.
+    /// </summary>
+    /// <param name="pProperty">The property to query.</param>
+    /// <returns>The current value of the property for this object.</returns>
+    int GetValue(const IntModelProperty& pProperty) const
+    {
+        IntModelProperty::ValueMap::const_iterator iter = m_mIntValues.find(pProperty.GetKey());
+        return (iter != m_mIntValues.end() ? iter->second : pProperty.GetDefaultValue());
+    }
+
+    /// <summary>
+    /// Sets the specified integer property to the specified value.
+    /// </summary>
+    /// <param name="pProperty">The property to set.</param>
+    /// <param name="nValue">The value to set.</param>
+    void SetValue(const IntModelProperty& pProperty, int nValue);
+
+private:
+    StringModelProperty::ValueMap m_mStringValues;
+    IntModelProperty::ValueMap m_mIntValues;
+
+#ifdef _DEBUG
+    // complete list of values as strings for viewing in the debugger
+    std::map<std::string, std::wstring> m_mDebugValues;
+#endif
+
+    typedef std::set<NotifyTarget*> NotifyTargetSet;
+    NotifyTargetSet m_vNotifyTargets;
```

Comment:
From https://en.cppreference.com/w/cpp/container/set/~set (applies to all polymorphic containers)
`...Note, that if the elements are pointers, the pointed-to objects are not destroyed.`
I'll suggest making `value_type ``std::unique_ptr<NotifyTarget>`.